Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -217,28 +219,14 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
         switch (assertionNode?.StatusVerification) {
           case GutterVerificationStatus.Verified: {
               var successDescription = description?.SuccessDescription ?? "_no message_";
-              if (successDescription == "error is impossible: This is the precondition that might not hold.") {
-                successDescription = "the precondition always holds";
-              }
               return $"{obsolescence}<span style='color:green'>**Success:**</span> " +
                      successDescription;
             }
           case GutterVerificationStatus.Error:
             var failureDescription = description?.FailureDescription ?? "_no message_";
-            if (currentlyHoveringPostcondition &&
-                  (failureDescription == new PostconditionDescription().FailureDescription ||
-                   failureDescription == new EnsuresDescription().FailureDescription)) {
-              failureDescription = "this postcondition might not hold on a return path";
-            }
-
-            if (!currentlyHoveringPostcondition &&
```

Comment:
Is it not possible to get the description that matches the diagnostic/related location of where the cursor is? Like, if you are hovering over the postcondition, can we not find the description that says "This is the postcondition that might not hold.", and if we hover of the return statement, the one that says "A postcondition might not hold on this return path." ?

Reply:
Actually, if hovering a postcondition was saying "This is the postcondition that might not hold.", it would feel weird to me, because it feels like there was a sentence referring to the postcondition before this one that this sentence answers. Otherwise, the error message would just say "This postcondition might not hold".
And this is actually the case. The message  "This is the postcondition that might not hold." was designed to appear as a relative error message in the diagnostics, just below "A postcondition might not hold on this return path".

This is why we have a different error message when hovering the postcondition by itself.

Now you could try to make a single error message for both cases, one that is relative to the error path, and the other one that is considering the postcondition only by itself. That message could be "This postcondition might not hold".
However, I'm considering the user's experience, I really liked "This is the postcondition that might not hold better" better, so that mean I had to specialize the other messagE.